samples/TagHelperSample.Web/Components/FeaturedMoviesComponent.cs

```diff
@@ -11,31 +13,38 @@ namespace TagHelperSample.Web.Components
     [ViewComponent(Name = "FeaturedMovies")]
     public class FeaturedMoviesComponent : ViewComponent
     {
-        private MoviesService _moviesService;
+        private readonly IMemoryCache _cache;
+        private readonly MoviesService _moviesService;
 
-        public FeaturedMoviesComponent(MoviesService moviesService)
+        public FeaturedMoviesComponent(MoviesService moviesService, IMemoryCache cache)
         {
             _moviesService = moviesService;
+            _cache = cache;
         }
 
         public IViewComponentResult Invoke()
         {
-            IExpirationTrigger trigger;
-            var movies = _moviesService.GetFeaturedMovies(out trigger);
-            
-            // Add custom triggers
-            EntryLinkHelpers.ContextLink.AddExpirationTriggers(new[] { trigger });
```

Comment:
cc @loudej \ @davidfowl 

This does take away the ability to add an arbitrary expiration trigger without caching it. I'm fine if this is the intended design for this, but it's different from what we'd initially discussed when designing the `CacheTagHelper`.